src/test/java/seedu/module/commons/util/StringUtilTest.java

```diff
@@ -13,33 +13,63 @@
     //---------------- Tests for isNonZeroUnsignedInteger --------------------------------------
 
     @Test
-    public void isNonZeroUnsignedInteger() {
+    public void isInteger() {
 
         // EP: empty strings
-        assertFalse(StringUtil.isNonZeroUnsignedInteger("")); // Boundary value
-        assertFalse(StringUtil.isNonZeroUnsignedInteger("  "));
+        assertFalse(StringUtil.isInteger("")); // Boundary value
+        assertFalse(StringUtil.isInteger("  "));
 
         // EP: not a number
-        assertFalse(StringUtil.isNonZeroUnsignedInteger("a"));
-        assertFalse(StringUtil.isNonZeroUnsignedInteger("aaa"));
+        assertFalse(StringUtil.isInteger("a"));
+        assertFalse(StringUtil.isInteger("aaa"));
+
+        // EP: numbers with white space
+        assertFalse(StringUtil.isInteger(" 10 ")); // Leading/trailing spaces
+        assertFalse(StringUtil.isInteger("1 0")); // Spaces in the middle
 
         // EP: zero
-        assertFalse(StringUtil.isNonZeroUnsignedInteger("0"));
+        assertTrue(StringUtil.isInteger("0"));
 
         // EP: zero as prefix
-        assertTrue(StringUtil.isNonZeroUnsignedInteger("01"));
+        assertTrue(StringUtil.isInteger("01"));
 
-        // EP: signed numbers
-        assertFalse(StringUtil.isNonZeroUnsignedInteger("-1"));
-        assertFalse(StringUtil.isNonZeroUnsignedInteger("+1"));
+        // EP: negative numbers
+        assertTrue(StringUtil.isInteger("-1"));
+        assertTrue(StringUtil.isInteger("-10"));
 
-        // EP: numbers with white space
-        assertFalse(StringUtil.isNonZeroUnsignedInteger(" 10 ")); // Leading/trailing spaces
-        assertFalse(StringUtil.isNonZeroUnsignedInteger("1 0")); // Spaces in the middle
+        // EP: positive signed numbers
+        assertTrue(StringUtil.isInteger("+1"));
+        assertTrue(StringUtil.isInteger("+10"));
+
+        // EP: positive unsigned numbers
+        assertTrue(StringUtil.isInteger("1")); // Boundary value
+        assertTrue(StringUtil.isInteger("10"));
+    }
+
+    @Test
+    public void isNonZeroUnsignedInteger() {
+
+        // EP: invalid inputs. Generate assertion error
```

Comment:
Is the reason here the method will directly detect that such string is not integer?

Reply:
The string should have been processed by isInteger() first so a non-integer would not be passed into isNonZeroUnsignedInteger()